iml-agent/src/action_plugins/check_ha.rs

```diff
@@ -38,14 +48,31 @@ fn create<'a>(elem: &Element, group: impl Into<Option<&'a str>>) -> ResourceAgen
 
 fn process_resource_list(output: &[u8]) -> Result<Vec<ResourceAgentInfo>, ImlAgentError> {
     let element = Element::from_reader(output)?;
-
+    let templates: HashMap<String, ResourceAgentType> = element
+        .find_all("template")
+        .map(|elem| {
+            (
+                elem.get_attr("id").unwrap_or("").to_string(),
+                ResourceAgentType::new(
+                    elem.get_attr("class"),
+                    elem.get_attr("provider"),
+                    elem.get_attr("type"),
+                ),
+            )
+        })
+        .collect();
     Ok(element
         .find_all("group")
         .flat_map(|g| {
+            let templates = &templates;
             g.find_all("primitive")
-                .map(move |p| create(p, g.get_attr("id").unwrap_or_default()))
+                .map(move |p| create(p, g.get_attr("id").unwrap_or_default(), templates))
```

Comment:
No need for `unwrap_or_default` here

```suggestion
                .map(move |p| create(p, g.get_attr("id"), templates))
```